src/definitions.jl

```diff
@@ -130,7 +130,14 @@ plan_fft
 """
     plan_fft!(A [, dims]; flags=FFTW.ESTIMATE, timelimit=Inf)
 
-Same as [`plan_fft`](@ref), but operates in-place on `A`.
+Same as [`plan_fft`](@ref),
+but creates a plan that operates in-place.
+`A` must be an array of complex floating-point numbers.
+When `p = plan_fft!(A, ...)`,
+one evaluates the FFT of an array `B`
+of the same `size`
+via `p * B` or via `mul!(B, p, B)`.
+For the `mul!` use, `B` must have the same `eltype` as `A`.
```

Comment:
Even for `p * B`, it won't work in-place if B doesn't have the same `eltype`, no?  

(Note also that the strides have to be the same too, when applying a plan to a new array.  This is typically only relevant for arrays that are views of other arrays, as otherwise the strides are determined by the size.)

Reply:
Actually, the `p * B` version works fine even if `B` has different `eltype` and `strides` from `A`:

```julia
using LinearAlgebra: mul!
using FFTW: plan_fft! 
dima = (8,16)
A = Array{ComplexF32}(undef, dima)
B = rand(ComplexF16, 4, dima...)
p = plan_fft!(A)
C = @view B[1,:,:] # strides and eltype differs from A
p * C # works fine!
D = collect(C)
@assert p * C == p * D # passes 
```

You make a good point that the `mul!` version requires matching `strides` too so I updated the PR to mention that.